modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -169,12 +173,26 @@
   @Inject
   protected ErrorTypeRepository errorTypeRepository;
 
+  @Inject
+  protected FeatureFlaggingService featureFlaggingService;
+
   private MetadataCacheIdGeneratorFactory<ComponentAst> cacheIdGeneratorFactory;
 
   protected MetadataCacheIdGenerator<ComponentAst> cacheIdGenerator;
 
   private Function<CoreEvent, Optional<ConfigurationInstance>> configurationResolver;
 
+  private static Class<ClassLoader> regionClClass;
+
+  static {
+    try {
+      regionClClass = (Class<ClassLoader>) ExtensionComponent.class.getClassLoader()
+          .loadClass("org.mule.runtime.module.artifact.api.classloader.RegionClassLoader");
```

Comment:
`ReginClassLaoder` is visible in this module, so you don't need to do this dynamic lookup